lib/core/application/event/event_application_form_bloc/event_application_form_bloc.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/domain/event/entities/event_application_question.dart';
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/utils/string_utils.dart';
+import 'package:app/graphql/backend/schema.graphql.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'event_application_form_bloc.freezed.dart';
+
+class EventApplicationFormBloc
+    extends Bloc<EventApplicationFormBlocEvent, EventApplicationFormBlocState> {
+  final List<EventApplicationQuestion>? initialQuestions;
+  EventApplicationFormBloc({this.initialQuestions})
+      : super(
+          EventApplicationFormBlocState(
+            fieldsState: {},
+            isValid: false,
+            answers: [],
+          ),
+        ) {
+    on<EventApplicationFormBlocEventInitFieldState>(initFieldState);
+    on<EventApplicationFormBlocEventUpdateField>(updateField);
+    on<EventApplicationFormBlocEventUpdateAnswer>(updateAnswer);
+  }
+
+  void initFieldState(
+    EventApplicationFormBlocEventInitFieldState event,
+    Emitter emit,
+  ) {
+    final user = event.user;
+    final applicationProfileFields =
+        event.event?.applicationProfileFields ?? [];
+    final applicationQuestions = event.event?.applicationQuestions ?? [];
+    final Map<String, String> initialFieldState = {};
+    final userMap = user?.toJson();
+    for (var applicationProfileField in applicationProfileFields) {
+      final key = applicationProfileField.field ?? '';
+      final value = userMap![StringUtils.snakeToCamel(key)];
+      initialFieldState[key] = value != null ? value.toString() : "";
```

Comment:
what if value is not a string here ? If you only support String ,then should check if the field value is String or not